.github/workflows/deb-release.yml

```
@@ -125,3 +137,70 @@ jobs:
         run: |
           gsutil -q -m rsync -r $HOME/.aptly/public/pool/main/z/zallet/ gs://${{ secrets.GCP_PROJECT_ID_PROD }}-apt-server/pool/main/z/zallet/
           gsutil -q -m rsync -r $HOME/.aptly/public/dists/ gs://${{ secrets.GCP_PROJECT_ID_PROD }}-apt-server/dists/
+
+      - name: Prepare Signature artifacts
+        run: |
+          mkdir artifacts
+          mv zallet_*.deb artifacts/
+          cd artifacts
+          gpg -u sysadmin@z.cash --armor --digest-algo SHA256 --detach-sign zallet_*.deb
+
+      # ==== Build Provenance Attestation (SLSA/Sigstore) ====
+      - name: Generate build provenance attestation for .deb
+        id: attest
+        uses: actions/attest-build-provenance@977bb373ede98d70efdf65b84cb5f73e068dcc2a
+        with:
+          subject-path: artifacts/*.deb
+
+      # === Make the attestation bundle filename unique per matrix ===
+      - name: Make attestation filename unique
+        shell: bash
+        env:
+          BUNDLE_PATH: ${{ steps.attest.outputs['bundle-path'] }}
+          MATRIX_NAME: ${{ matrix.name }}
+        run: |
+          set -Eeuo pipefail
+          IFS=$'\n\t'
+      
+          mkdir -p artifacts
+      
+          # (defense-in-depth) allow only safe chars in MATRIX_NAME
+          case "$MATRIX_NAME" in
+            (*[!a-zA-Z0-9._-]*|'') echo "Invalid matrix name: $MATRIX_NAME" >&2; exit 2;;
+          esac
+      
+          # Use quotes and `--` to avoid argument injection/word splitting
+          cp -- "$BUNDLE_PATH" "artifacts/attestation-${MATRIX_NAME}.intoto.jsonl"
```

Per the [naming convention](https://github.com/in-toto/attestation/blob/main/spec/v1/bundle.md#file-naming-convention):
- The suffix is correct.
- The prefix should be something more relevant to what the attestation covers.

Given that each entry in the matrix produces a single .deb file, we should just follow the naming convention precisely, and name this `<filename>.intoto.jsonl`:
```suggestion
          # Each matrix strategy builds a single package, so this matches a single file.
          FILENAME=$(basename artifacts/zallet_*.deb)

          # Use quotes and `--` to avoid argument injection/word splitting
          cp -- "$BUNDLE_PATH" "artifacts/${FILENAME}.intoto.jsonl"
```

This also handles the fact that we already renamed the .deb file earlier to not collide across the matrix.